WireDomain/Sources/WireDomain/Repositories/Conversations/ConversationLocalStore+Group.swift

```diff
@@ -0,0 +1,100 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import WireAPI
+import WireDataModel
+
+/// An extension that encapsulates storage operations related to conversation group (user, role, members).
+
+extension ConversationLocalStore {
+
+    // MARK: - User & Role
+
+    func fetchUserAndRole(
+        from remoteConversationMember: WireAPI.Conversation.Member,
+        for localConversation: ZMConversation
+    ) -> (user: ZMUser, role: Role?)? {
+        guard let userID = remoteConversationMember.id ?? remoteConversationMember.qualifiedID?.uuid else {
+            return nil
+        }
+
+        let user = ZMUser.fetchOrCreate(
+            with: userID,
+            domain: remoteConversationMember.qualifiedID?.domain,
+            in: context
+        )
+
+        func fetchOrCreateRoleForConversation(
+            name: String,
+            conversation: ZMConversation
+        ) -> Role {
+            Role.fetchOrCreateRole(
+                with: name,
+                teamOrConversation: conversation.team != nil ? .team(conversation.team!) : .conversation(conversation),
```

Comment:
there's a helper method in DataModel for this
```suggestion
                teamOrConversation: .fromTeamOrConversation(team: team, conversation: conversation),
```

Reply:
fixed, we can actually use the public static func `matching(_ conversation: ZMConversation) -> TeamOrConversation` but it does call the method you suggested. thanks.